src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/StatefulMarshallingStrategy.cs

```diff
@@ -444,8 +444,28 @@ public IEnumerable<StatementSyntax> GenerateUnmarshalStatements(StubIdentifierCo
 
             string numElementsIdentifier = MarshallerHelpers.GetNumElementsIdentifier(TypeInfo, context);
 
+            // Generate numElements expression with null check for pointer types
+            ExpressionSyntax numElementsExpression = ElementsMarshalling.GenerateNumElementsExpression(countInfo, castCountInfo, CodeContext, context);
+
+            // If the marshalling direction is unmanaged-to-managed and we have native pointers,
+            // we need to check if the native pointer is null before using the size parameter to avoid
+            // allocating arrays for null pointers. This applies to LPArray scenarios.
+            if (CodeContext.Direction == MarshalDirection.UnmanagedToManaged &&
+                innerMarshaller.NativeType.FullTypeName.EndsWith("*"))
```

Comment:
```suggestion
                innerMarshaller.NativeType.FullTypeName.EndsWith("*"))
```

@jkoritzinsky or @jtschuster This seems like a suspect check. Is there a better way to check for array scenarios?

Reply:
@copilot please try to apply the suggestion by @jtschuster